src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -80,6 +88,27 @@
 
     private Map<String,String> descriptions = new ConcurrentHashMap<String, String>();
 
+    /**
+     * Tool tips for "Overall Build Analysis" trend type.
+     */
+    private Map<NumberOnlyBuildLabel,String> failToolTip, totalToolTip, skipToolTip;
+
+    /**
+     * Tool tips for "Lengthy Tests" trend type.
+     */
+    private Map<NumberOnlyBuildLabel, String> lengthyToolTip;
+
+    /**
+     * Array for storing package hierarchy derived from each testcase, used for selecting a particular
+     * project.
+     */
+    private String[] projectList;
```

Comment:
We have fixed this to size 50, right ? You can initialize same here, Or better you can use maxHeap to store.